resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceRequestHelperSuite.scala

```diff
@@ -0,0 +1,255 @@
+    ResourceRequestTestHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "123ppp")
+    val resource = createAResource
+
+    val thrown = intercept[IllegalArgumentException] {
+      ResourceRequestHelper.setResourceRequests(resourceTypes, resource)
+    }
+    thrown.getMessage should fullyMatch regex
+      """Unknown unit 'ppp'\. Known units are \[.*\]"""
+  }
+
+  test("resource type with valid value and without unit") {
+    assume(ResourceRequestHelper.isYarnResourceTypesAvailable())
+    ResourceRequestTestHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "123")
+    val resource = createAResource
+
+    ResourceRequestHelper.setResourceRequests(resourceTypes, resource)
+    val resourceInfo: ResourceInformation = ResourceRequestTestHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    assert(resourceInfo === ResourceInformation(CUSTOM_RES_1, 123, ""))
+  }
+
+  test("resource type with valid value and unit") {
+    assume(ResourceRequestHelper.isYarnResourceTypesAvailable())
+    ResourceRequestTestHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "2g")
+    val resource = createAResource
+
+    ResourceRequestHelper.setResourceRequests(resourceTypes, resource)
+    val resourceInfo: ResourceInformation = ResourceRequestTestHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    assert(resourceInfo === ResourceInformation(CUSTOM_RES_1, 2, "G"))
+  }
+
+  test("two resource types with valid values and units") {
+    assume(ResourceRequestHelper.isYarnResourceTypesAvailable())
+    ResourceRequestTestHelper.initializeResourceTypes(List(CUSTOM_RES_1, CUSTOM_RES_2))
+
+    val resourceTypes = Map(
+      CUSTOM_RES_1 -> "123m",
+      CUSTOM_RES_2 -> "10G"
+    )
+    val resource = createAResource
+
+    ResourceRequestHelper.setResourceRequests(resourceTypes, resource)
+    val resourceInfo: ResourceInformation = ResourceRequestTestHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    assert(resourceInfo === ResourceInformation(CUSTOM_RES_1, 123, "m"))
+
+    val resourceInfo2: ResourceInformation = ResourceRequestTestHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_2)
+    assert(resourceInfo2 === ResourceInformation(CUSTOM_RES_2, 10, "G"))
+  }
+
+  test("empty SparkConf should be valid") {
+    val sparkConf = new SparkConf()
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("just normal resources are defined") {
+    val sparkConf = new SparkConf()
+    sparkConf.set(DRIVER_MEMORY.key, "3G")
+    sparkConf.set(DRIVER_CORES.key, "4")
+    sparkConf.set(EXECUTOR_MEMORY.key, "4G")
+    sparkConf.set(EXECUTOR_CORES.key, "2")
+
+    ResourceRequestHelper.validateResources(sparkConf)
+  }
+
+  test("Memory defined with new config for executor") {
```

Comment:
nit: either start all test descriptions with capital letters or not. I prefer to not use capital letters here.

All these validation tests could use a helper function since they're all extremely similar. You can use something like:

```
listOfThingsToTest.foreach { blah =>
  test(s"validate $blah") {
     // test code
   }
}
```

Reply:
As stated in another comment, I extracted the common parts of the method in this test class. Also fixed the naming to be consistent with each other.